akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -86,6 +82,53 @@ import akka.util.Helpers.toRootLowerCase
     (bytes(0) == GZIPInputStream.GZIP_MAGIC.toByte) &&
     (bytes(1) == (GZIPInputStream.GZIP_MAGIC >> 8).toByte)
   }
+
+  case class LZ4Meta(offset: Int, length: Int) {
+    import LZ4Meta._
+
+    def putInto(buffer: ByteBuffer): Unit = {
+      buffer.putInt(LZ4_MAGIC)
+      buffer.putInt(length)
+    }
+
+    def prependTo(bytes: Array[Byte]): Array[Byte] = {
+      val buffer = ByteBuffer.allocate(bytes.length + offset)
+      putInto(buffer)
+      buffer.put(bytes)
+      buffer.array()
```

Comment:
Ok, so this actually returns the backing array, so it's just the one allocate+arraycopy plus some extra, with some luck the ByteBuffer instance is escape-analyzed away. 👍

Reply:
`buffer.array` is not safe to use in general, but here it should be fine since we know it's a HeapByteBuffer